crates/miden-lib/asm/kernels/transaction/main.masm

```diff
@@ -129,27 +130,31 @@ proc.main.1
     # get the memory address of the transaction script root and load it to the stack
     exec.memory::get_tx_script_root_ptr
     padw dup.4 mem_loadw
-    # => [TX_SCRIPT_ROOT, tx_script_root_ptr]
+    # => [TX_SCRIPT_ROOT, tx_script_root_ptr, pad(16)]
 
     exec.utils::is_empty_word not movdn.4 dropw
-    # => [has_tx_script, tx_script_root_ptr]
+    # => [has_tx_script, tx_script_root_ptr, pad(16)]
 
     if.true
+        # pad the stack
+        padw padw swapdw push.0 swap push.0 swap push.0 swap
```

Comment:
nit: instead of `push.0 swap push.0 swap push.0 swap` could we do `push.0.0.0 movup.3`?